### ISSUE TITLE
FieldDescriptorFactoryInterface not available in webspace context

### ISSUE THREAD

**Issue by axzx** [sulu/sulu#4946]
| Q | A
| --- | ---
| Bug? | yes
| New Feature? | no
| Sulu Version | 2.0.2

#### Actual Behavior

```
Cannot autowire service "App\Controller\Admin\PointController": argument "$fieldDescriptorFactory" of method "__construct()" references interface "Sulu\Component\Rest\ListBuilder\Metadata\FieldDescriptorFactoryInterface" but no such service exists. Did you create a class that implements this interface?
```
### Expected Behavior

working list in admin

#### Steps to Reproduce

http://docs.sulu.io/en/2.0/book/extend-admin.html

#### Possible Solutions

If you have already ideas how to solve the issue, add them here.
(remove this section if not needed)

**Comment by danrot** [comment]
I think that is the same issue as #4399. The solution is to currently move your controller to the `App\Controller\Admin` namespace, because then we only register this controller in the admin context, not on the website. Seems like I didn't update the documentation accordingly, will do that now.

**Comment by axzx** [comment]
> The solution is to currently move your controller to the `App\Controller\Admin` namespace

but I already have it done

Cannot autowire service **"App\Controller\Admin\PointController"**


I can't see routes with my PointController in `./bin/adminconsole debug:router |grep point`

**Comment by danrot** [comment]
Have you also moved the controller to the `src/Controller/Admin` folder?

**Comment by axzx** [comment]
![Zrzut ekranu z 2019-12-13 10-57-30](https://user-images.githubusercontent.com/1436016/70791638-bf00f680-1d97-11ea-9497-9f47adef44dd.png)

**Comment by danrot** [comment]
Do you also have the following lines in your project?

https://github.com/sulu/skeleton/blob/7ca7104a9b73ea5a8d1244e30bd3c13d663ec5cb/config/services.yaml#L24-L33

**Comment by axzx** [comment]
> Do you also have the following lines in your project?
> 
> https://github.com/sulu/skeleton/blob/7ca7104a9b73ea5a8d1244e30bd3c13d663ec5cb/config/services.yaml#L24-L33

No, I didn't know that. 
I created the project on December 6.

but adding it did nothing
still does'n work

**Comment by axzx** [comment]
maybe I should create the project from scratch on the current skeleton?

**Comment by danrot** [comment]
Would be worth a try, then we know that the problem is with your current setup...

**Comment by NyoCon** [comment]
I got the same Problem.

I did the following:

- create a new project with composer create-project sulu/skeleton sulu -n
- change DATABASE_URL
- run sulu:build dev
- create a TestController in App\Controller\Admin with the following content:

```
<?php

namespace App\Controller\Admin;

use FOS\RestBundle\View\ViewHandlerInterface;
use Symfony\Component\HttpFoundation\Response;
use Symfony\Component\Routing\Annotation\Route;

class ArticleController
{
    public function __construct(ViewHandlerInterface $viewHandler)
    {
    }

    /**
     * @Route("/test")
     */
    public function test()
    {
        return new Response('OK');
    }
}
```
- start server with bin/console server:start
- navigate to /test in browser
- get the following error:

```
The controller for URI "/test" is not callable. Controller "App\Controller\Admin\ArticleController" cannot be fetched from the container because it is private. Did you forget to tag the service with "controller.service_arguments"?
```
Context entries in services.yaml are there.

**Comment by axzx** [comment]
I created a new project based on the instructions
http://docs.sulu.io/en/2.0/book/getting-started.html

I created the controller based on the instructions
http://docs.sulu.io/en/2.0/book/extend-admin.html

`./bin/adminconsole debug:router |grep event`
I don't see any routes

**Comment by alexander-schranz** [comment]
@axzx @NyoCon if you are extending a admin, your route need to start with `/admin/..` else it is not callable in the admin context. Also make sure you cleared the cache (bin/adminconsole cache:clear) as symfony is caching routes on container build.

**Comment by axzx** [comment]
@alexander-schranz I do not understand.
Is this somewhere described on this page http://docs.sulu.io/en/2.0/book/extend-admin.html?

**Comment by alexander-schranz** [comment]
@axzx you need to register the routes as written before http://docs.sulu.io/en/2.0/book/extend-admin.html it depends on how you build your controller. Sulu itself doesn't force you how you build your rest controllers if using the fos rest bundle with the ClassResourceInterface on the controller you can register the controller the following way in your `config/routes_admin.yaml` e.g.:

```
app_routes:
    type: rest
    prefix: /admin/api
    resource: App\Controller\Admin\EventController
    name_prefix: app.
```

if you want to register routes with annotations use annotations. But make sure its callable from the admin context as you see in your [public/index.php](https://github.com/sulu/skeleton/blob/ea1166435dbe555d0c01db14007f534c0b022d04/public/index.php#L45) only routes with `/admin/...` are callable in the admin context.

**Comment by danrot** [comment]
The solution from Alex should be working, but IMO we should still extract that part into a separate Bundle, so that the `FieldDescriptorFactory` can also be used in the website context.

**Comment by nnatter** [comment]
I think the unsolved part of this is equal to #4399. One of them can probably be closed 🙂

**Comment by danrot** [comment]
Closed #4399 since this issue is more active.

**Comment by niklasnatter** [comment]
Fixed by #5942
